### PR TITLE
Improves notifications for contribution

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/2114

Native implementation: https://github.com/brave-intl/bat-native-ledger/pull/179

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
1. 
- enable rewards
- claim grant
- do a tip
- make sure that you see notification with correct amount

image

2.
(start with reconcile interval flag)
- enable rewards
- claim grant
- set monthly amount to 50
- add site to ac table
- make sure that you see notification telling you that there was an error because you don't have enough funds

image

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: components/brave_rewards/resources/extension/brave_rewards/_locales/en_GB/messages.json [hunk 1]
```diff
@@ -168,11 +168,11 @@
     "description": ""
   },
   "braveContributeTitle": {
-    "message": "Contribute",
```
Comment (bbondy) [inline components/brave_rewards/resources/extension/brave_rewards/_locales/en_GB/messages.json]: everything except for the en_US messages.json file is auto-generated, so pls remove changes to this file.

File: components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx [hunk 2]
```diff
@@ -191,6 +191,8 @@ export class Panel extends React.Component<Props, State> {
 
     const notification: RewardsExtension.Notification = notifications[currentNotification]
 
+    console.log(notification)
```
Comment (bbondy) [inline components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: remove console.log

File: components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx [hunk 3]
```diff
@@ -202,10 +204,18 @@ export class Panel extends React.Component<Props, State> {
...
-            text = getMessage('contributeNotificationSuccess')
+          const result = notification.args[1]
+
+          if (result === '0') {
+            const fixed = utils.convertProbiToFixed(notification.args[3])
+            text = getMessage('contributeNotificationSuccess',
+              [fixed])
           } else {
-            text = getMessage('contributeNotificationError')
+            if (result === '14') {
```
Comment (bbondy) [inline components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: what is this '14' and '15'? Can we add a comment or use a constant here?

Comment (bbondy) [reply components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: you can probably just move this into else if branches above

File: components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json [hunk 4]
```diff
@@ -188,5 +188,17 @@
   "braveRewardsCreatingText": {
     "message": "Creating wallet",
     "description": ""
+  },
+  "contributeNotificationError": {
+    "message": "There was a problem processing your contribution.",
+    "description": ""
```
Comment (bbondy) [inline components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json]: Please add descriptions for what the string is for.

File: components/brave_rewards/resources/extension/brave_rewards/_locales/en_GB/messages.json (hunk 1 above)
Comment (NejcZdovc) [reply components/brave_rewards/resources/extension/brave_rewards/_locales/en_GB/messages.json]: removed

File: components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json (hunk 4 above)
Comment (NejcZdovc) [reply components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json]: added

File: components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx (hunk 2 above)
Comment (NejcZdovc) [reply components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: removed

File: components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx [hunk 5]
```diff
@@ -202,10 +202,23 @@ export class Panel extends React.Component<Props, State> {
...
+
+          if (result === '0') {
+            const fixed = utils.convertProbiToFixed(notification.args[3])
+            text = getMessage('contributeNotificationSuccess',
+              [fixed])
+          } else if (result === '14') {
+            text = getMessage('contributeNotificationEmptyTable')
+          } else if (result === '15') {
+            text = getMessage('contributeNotificationNotEnoughFunds')
           }
```
Comment (bbondy) [inline components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: Why don't we handle all available errors here from the table? Or at least have an else clause for if the error is outside of those.

File: components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx [hunk 6]
```diff
@@ -202,10 +202,23 @@ export class Panel extends React.Component<Props, State> {
...
+          // 16 - error while tipping
+
+          if (result === '0') {
+            const fixed = utils.convertProbiToFixed(notification.args[3])
+            text = getMessage('contributeNotificationSuccess',
+              [fixed])
+          } else if (result === '14') {
+            text = getMessage('contributeNotificationEmptyTable')
+          } else if (result === '15') {
+            text = getMessage('contributeNotificationNotEnoughFunds')
```
Comment (bbondy) [inline components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: Why don't we handle all available errors from the above table here? Or at least have an else clause to catch other ones?

Comment (NejcZdovc) [reply components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: done

**Comment by NejcZdovc** [comment]
master (0.59) bc3ac42
0.58.x e26f4ad6
0.57.x 57f93eae

**Pull request merged**